js/main.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+{
+	const URL = 'https://opentdb.com/api.php?amount=10&type=multiple';
+	const main = document.querySelector('main');
+	let questionIndex = 0;
+	let correct = 0;
+
+	function deleteMain() {
+		while (main.firstChild) {
+			main.removeChild(main.firstChild);
+		}
+	}
+
+	class BasicPage {
+		constructor() {
+			this.title = document.createElement('h2');
+			main.appendChild(this.title);
+
+			this.statement = document.createElement('p');
+			this.statement.id = 'statement';
+			main.appendChild(this.statement);
+		}
+	}
+
+	class FirstPage {
+		constructor(basicPage) {
+			this.basicPage = basicPage;
+			this.getText();
+			this.makeBtn();
+		}
+
+		getText() {
+			this.basicPage.title.textContent = 'ようこそ';
+			this.basicPage.statement.textContent = '以下のボタンをクリック';
+		}
+
+		makeBtn() {
+			this.btn = document.createElement('button');
+			this.btn.textContent = '開始';
+	
+			this.btn.addEventListener('click', () => {
+				this.btnFnc();
+			});
+			main.appendChild(this.btn);
+		}
+		
+		btnFnc() {
+			deleteMain();
+			new LoadingPage(new BasicPage());
+	
+			fetch(URL)
+				.then((response) => response.json())
+				.then((json) => {
+					const questions = json.results;
+					deleteMain();
+					new Question(new BasicPage(), questions);
+			});
```

Comment:
`async` `await` と `try/catch` を使うとよりよいと思いました。

- `async` `await`
  - Javascript 独自の Promise を意識しなくてよくなります
  - 記述が他言語と近くなり、見やすくなります
- `try/catch`
  - API アクセスなどは他の箇所よりエラーが発生しやすいです(ネットワークエラー等)
  - このようにエラーハンドリングすることで、想定外のエラーに対処しやすくなります

```suggestion
			try {
				const response = await fetch(URL);
				const json = await response.json();
				const questions = json.results;
				deleteMain();
				new Question(new BasicPage(), questions);
			} catch (e) {
				console.error(e);
			}
```

Reply:
ありがとうございます。
確かにこの方がスッキリしますね。